contracts/src/periphery/cfg-getter/EigenDAConfigRetriever.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDAConfigRegistry} from "src/core/interfaces/IEigenDADirectory.sol";
+import {ConfigRegistryTypes} from "src/core/libraries/v3/config-registry/ConfigRegistryTypes.sol";
+import {ConfigRegistryLib} from "src/core/libraries/v3/config-registry/ConfigRegistryLib.sol";
+
+/// @title EigenDAConfigRetriever
+/// @notice A stateless contract that defines getter functions using the underlying configuration registry's primitive API.
+contract EigenDAConfigRetriever {
+    IEigenDAConfigRegistry public immutable configRegistry;
+
+    constructor(address configRegistryAddress) {
+        configRegistry = IEigenDAConfigRegistry(configRegistryAddress);
+    }
+
+    /// @notice Retrieves the currently active bytes32 configuration checkpoint and all future checkpoints for a given name.
+    /// @dev Returns the checkpoint with the highest activation key that is less than or equal to the provided activation key,
+    ///      plus all checkpoints with activation keys greater than the provided activation key.
+    ///      This allows clients to know the current configuration value and plan ahead for upcoming updates.
+    function getActiveAndFutureBytes32Configs(string memory name, uint256 activationKey)
+        external
+        view
+        returns (ConfigRegistryTypes.Bytes32Checkpoint[] memory)
+    {
+        bytes32 nameDigest = ConfigRegistryLib.getNameDigest(name);
+        uint256 numCheckpoints = configRegistry.getNumCheckpointsBytes32(nameDigest);
+
+        // There are 3 cases to handle:
+        // 1. If no checkpoints have activation keys less than or equal to the provided activation key, we return an empty array.
```

Comment:
I know you just carried this logic forward, but IMO this isn't the behavior we actually want
```suggestion
        // 1. If no checkpoints have activation keys less than or equal to the provided activation key, we return all checkpoints as future values.
```

I don't know why you'd want it to return empty if there isn't any active value. The idea is that we should see future updates coming in advance, so it makes sense you'd just return all future values even if there isn't any active value